Creating a new `Exception` without actually throwing it is useless and is probably due to a mistake.

## Noncompliant Code Example

<pre>
if (x &lt; 0)
{
  new ArgumentException("x must be nonnegative");
}
</pre>

## Compliant Solution

<pre>
if (x &lt; 0)
{
  throw new ArgumentException("x must be nonnegative");
}
</pre>