The best way to determine the type of a generic method is by inference based on the type of argument that is passed to the method. This is not
possible when a parameter type is missing from the argument list.

## Noncompliant Code Example

<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void MyMethod
 <t>
  ()  // Noncompliant
    {
    }
  }
}

 </t></pre>

## Compliant Solution

<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void MyMethod
 <t>
  (T param)
    {
    }
  }
}

 </t></pre>