Changing an inherited member to `private` will not prevent access to the base class implementation.

This rule raises an issue when a `private`, non-`final` method in an unsealed type has a signature that is identical to a
`public` method declared in a base type.

## Noncompliant Code Example

<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public class Bar:Foo
  {
    private void SomeMethod(int count) { } // Noncompliant
  }
}
</pre>

## Compliant Solution

<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public class Bar:Foo
  {
    private final void SomeMethod(int count) { }
  }
}
</pre>