Certain characters, once normalized to lowercase, cannot make a round trip, meaning that they can not be converted from one locale to another and
then accurately restored to their original characters.

It is therefore strongly recommended to normalize strings to uppercase instead.

## Noncompliant Code Example

<pre>
String s = myString.ToLower(CultureInfo.InvariantCulture);  // Noncompliant
</pre>

## Compliant Solution

<pre>
String s = myString.ToUpper(CultureInfo.InvariantCulture);
</pre>