Merging collapsible `if` statements increases the codeâ€™s readability.
 
## Noncompliant Code Example

    if (condition1)
    {
        if (condition2)
        {
            // ...
        }
    }

## Compliant Solution

    if (condition1 && condition2)
    {
        // ...
    }