
Merging collapsible `if` statements increases the code's readability.

## Noncompliant Code Example


    if (condition1)
    {
        if (condition2)
        {
            // ...
        }
    }


## Compliant Solution


    if (condition1 &amp;&amp; condition2)
    {
        // ...
    }

