The complexity of an expression is defined by the number of `&&`, `||` and `condition ? ifTrue : ifFalse`
operators it contains.

A single expression's complexity should not become too high to keep the code readable.

## Noncompliant Code Example

With the default threshold value of 3

<pre>
if (((condition1 &amp;&amp; condition2) || (condition3 &amp;&amp; condition4)) &amp;&amp; condition5) { ... }
</pre>

## Compliant Solution

<pre>
if ((MyFirstCondition() || MySecondCondition()) &amp;&amp; MyLastCondition()) { ... }
</pre>