Merging collapsible `if` statements increases the code's readability.

## Noncompliant Code Example

<pre>
if (condition1)
{
  if (condition2)
  {
    ...
  }
}
</pre>

## Compliant Solution

<pre>
if (condition1 &amp;&amp; condition2)
{
  ...
}
</pre>