Most of the time a block of code is empty when a piece of code is really missing. So such empty block must be either filled or removed.

## Noncompliant Code Example

<pre>
for (int i = 0; i &lt; 42; i++){}  // Empty on purpose or missing piece of code ?
</pre>

## Exceptions

When a block contains a comment, this block is not considered to be empty.