
Shared coding conventions make it possible for a team to efficiently collaborate. This rule makes it mandatory to place a close curly brace at the<br>beginning of a line.

## Noncompliant Code Example


    if(condition)
    {
      doSomething();}


## Compliant Solution


    if(condition)
    {
      doSomething();
    }


## Exceptions

When blocks are inlined (open and close curly braces on the same line), no issue is triggered.


    if(condition) {doSomething();}

