Shared coding conventions make it possible for a team to efficiently collaborate. This rule makes it mandatory to place a close curly brace at the
beginning of a line.

## Noncompliant Code Example

<pre>
if(condition)
{
  doSomething();}
</pre>

## Compliant Solution

<pre>
if(condition)
{
  doSomething();
}
</pre>

## Exceptions

When blocks are inlined (open and close curly braces on the same line), no issue is triggered. 

<pre>
if(condition) {doSomething();}
</pre>