Overriding or shadowing a variable declared in an outer scope can strongly impact the readability, and therefore the maintainability, of a piece of code. Further, it could lead maintainers to introduce bugs because they think theyâ€™re using one variable but are really using another.
 
## Noncompliant Code Example

    class Foo
    {
      public int myField;
    
      public void DoSomething()
      {
        int myField = 0;  // Noncompliant
        ...
      }
    }