## Why is this an issue?
 
This rule is meant to be used as a way to track code which is marked as being deprecated. Deprecated code should eventually be removed.
 
### Noncompliant code example

    [Obsolete] // Noncompliant
    void Method()
    {
        // ..
    }