
While not technically incorrect, the omission of curly braces can be misleading, and may lead to the introduction of errors during maintenance.

## Noncompliant Code Example


    // the two statements seems to be attached to the if statement, but that is only true for the first one:
    if (condition)
      ExecuteSomething();
      CheckSomething();


## Compliant Solution


    if (condition)
    {
      ExecuteSomething();
      CheckSomething();
    }


## See

- [CERT, EXP19-C.](https://wiki.sei.cmu.edu/confluence/x/g9YxBQ) - Use braces for the body of an if, for, or while statement
- [CERT, EXP52-J.](https://wiki.sei.cmu.edu/confluence/x/MzZGBQ) - Use braces for the body of an if, for, or while statement

