
When you implement `IComparable` or `IComparable<T>` on a class you should also override `Equals(object)`<br>and overload the comparison operators (`==`, `!=`, `<`, `<=`, `>`,<br>`>=`). That's because the CLR cannot automatically call your `CompareTo` implementation from `Equals(object)` or<br>from the base comparison operator implementations. Additionally, it is best practice to override `GetHashCode` along with<br>`Equals`.

This rule raises an issue when a class implements `IComparable` without also overriding `Equals(object)` and the comparison<br>operators.

## Noncompliant Code Example


    public class Foo: IComparable  // Noncompliant
    {
      public int CompareTo(object obj) { /* ... */ }
    }


## Compliant Solution


    public class Foo: IComparable
    {
      public int CompareTo(object obj) { /* ... */ }
      public override bool Equals(object obj)
      {
        var other = obj as Foo;
        if (object.ReferenceEquals(other, null))
        {
          return false;
        }
        return this.CompareTo(other) == 0;
      }
      public int GetHashCode() { /* ... */ }
      public static bool operator == (Foo left, Foo right)
      {
        if (object.ReferenceEquals(left, null))
        {
          return object.ReferenceEquals(right, null);
        }
        return left.Equals(right);
      }
      public static bool operator &gt; (Foo left, Foo right)
      {
        return Compare(left, right) &gt; 0;
      }
      public static bool operator &lt; (Foo left, Foo right)
      {
        return Compare(left, right) &lt; 0;
      }
      public static bool operator != (Foo left, Foo right)
      {
        return !(left == right);
      }
    }

