`break;` is an unstructured control flow statement which makes code harder to read.
 
Ideally, every loop should have a single termination condition.
 
## Noncompliant Code Example

    int i = 0;
    while (true)
    {
      if (i == 10)
      {
        break;      // Non-Compliant
      }
    
      Console.WriteLine(i);
      i++;
    }

## Compliant Solution

    int i = 0;
    while (i != 10) // Compliant
    {
      Console.WriteLine(i);
      i++;
    }