
When only the condition expression is defined in a `for` loop, and the initialization and increment expressions are missing, a<br>`while` loop should be used instead to increase readability.

## Noncompliant Code Example


    for (;condition;) { /*...*/ }


## Compliant Solution


    while (condition) { /*...*/ }

