
`switch` statements are useful when there are many different cases depending on the value of the same expression.

For just one or two cases however, the code will be more readable with `if` statements.

## Noncompliant Code Example


    switch (variable)
    {
      case 0:
        doSomething();
        break;
      default:
        doSomethingElse();
        break;
    }


## Compliant Solution


    if (variable == 0)
    {
      doSomething();
    }
    else
    {
      doSomethingElse();
    }

