If a local variable is declared but not used, it is dead code and should be removed. Doing so will improve maintainability because developers will not wonder what the variable is used for.
 
## Noncompliant Code Example

    public int NumberOfMinutes(int hours)
    {
      int seconds = 0;   // seconds is never used
      return hours * 60;
    }

## Compliant Solution

    public int NumberOfMinutes(int hours)
    {
      return hours * 60;
    }

## Exceptions
 
Unused locally created resources in a `using` statement are not reported.

    using(var t = new TestTimer()) // t never used, but compliant.
    {
      //...
    }