
There is no good reason to create a new object to not do anything with it. Most of the time, this is due to a missing piece of code and so could<br>lead to an unexpected behavior in production.

If it was done on purpose because the constructor has side-effects, then that side-effect code should be moved into a separate, static method and<br>called directly.

## Noncompliant Code Example


    if (x &lt; 0)
      new ArgumentException("x must be nonnegative");


## Compliant Solution


    if (x &lt; 0)
      throw new ArgumentException("x must be nonnegative");

