
A chain of `if`/`else if` statements is evaluated from top to bottom. At most, only one branch will be executed: the first<br>one with a condition that evaluates to `true`.

Therefore, duplicating a condition automatically leads to dead code. Usually, this is due to a copy/paste error. At best, it's simply dead code and<br>at worst, it's a bug that is likely to induce further bugs as the code is maintained, and obviously it could lead to unexpected behavior.

## Noncompliant Code Example


    if (param == 1)
    {
      OpenWindow();
    }
    else if (param == 2)
    {
      CloseWindow();
    }
    else if (param == 1) // Noncompliant
    {
      MoveWindowToTheBackground();
    }


## Compliant Solution


    if (param == 1)
    {
      OpenWindow();
    }
    else if (param == 2)
    {
      CloseWindow();
    }
    else if (param == 3)
    {
      MoveWindowToTheBackground();
    }


## See

- [CERT, MSC12-C.](https://www.securecoding.cert.org/confluence/x/NYA5) - Detect and remove code that has no effect or is never<br>  executed

