
It can be extremely confusing when a `for` loop's counter is incremented outside of its increment clause. In such cases, the increment<br>should be moved to the loop's increment clause if at all possible.

## Noncompliant Code Example


    for (i = 0; i &lt; 10; j++) // Noncompliant
    {
      // ...
    }


## Compliant Solution


    for (i = 0; i &lt; 10; i++)
    {
      // ...
    }

