## Why is this an issue?
 
There is no good excuse for an empty class. If itâ€™s being used simply as a common extension point, it should be replaced with an `interface`. If it was stubbed in as a placeholder for future development it should be fleshed-out. In any other case, it should be eliminated.
 
### Noncompliant code example

    public class Empty // Noncompliant
    {
    }

### Compliant solution

    public interface IEmpty
    {
    }

### Exceptions
 
Partial classes are ignored entirely, as they are often used with Source Generators. Subclasses of System.Exception are ignored, as even an empty Exception class can provide useful information by its type name alone. Subclasses of System.Attribute are ignored, as well as classes which are annotated with attributes. Subclasses of generic classes are ignored, as even when empty they can be used for type specialization. Subclasses of certain framework types - like the PageModel class used in ASP.NET Core Razor Pages - are also ignored.

    using Microsoft.AspNetCore.Mvc.RazorPages;
    
    public class EmptyPageModel: PageModel // Compliant - an empty PageModel can be fully functional, the C# code can be in the cshtml file
    {
    }