When the modulus of a negative number is calculated, the result will either be negative or zero. Thus, comparing the modulus of a variable for
equality with a positive number (or a negative one) could result in unexpected results. 

## Noncompliant Code Example

<pre>
public bool IsOdd(int x)
{
  return x % 2 == 1;  // Noncompliant; if x is an odd negative, x % 2 == -1
}
</pre>

## Compliant Solution

<pre>
public bool IsOdd(int x)
{
  return x %2 != 0;
}
</pre>

or

<pre>
public bool IsOdd(uint x)
{
  return x %2 == 1;
}
</pre>

## See

*   [CERT, NUM51-J.](https://www.securecoding.cert.org/confluence/x/xAHAAQ) - Do not assume that the remainder operator always returns a
      nonnegative result for integral operands
*   [CERT, INT10-C](https://www.securecoding.cert.org/confluence/x/NQBi) - Do not assume a positive remainder when using the % operator

*   [CERT, INT10-CPP.](https://www.securecoding.cert.org/confluence/x/_YBLAQ) - Do not assume a positive remainder when using the %
      operator