Debug statements are always useful during development. But include them in production code - particularly in code that runs client-side - and you
run the risk of inadvertently exposing sensitive information.

## Noncompliant Code Example

<pre>
private void DoSomething ()
{
  // ...
  Console.WriteLine ("so far, so good..."); // Noncompliant
  // ...
}
</pre>

## See

*   [OWASP Top Ten 2013 Category A6](https://www.owasp.org/index.php/Top_10_2013-A6-Sensitive_Data_Exposure) - Sensitive Data Exposure