
Because composite format strings are interpreted at runtime, rather than validated by the compiler, they can contain errors that lead to unexpected<br>behaviors or runtime errors. This rule statically validates the good behavior of composite formats when calling the methods of<br>`String.Format`, `StringBuilder.AppendFormat`, `Console.Write`, `Console.WriteLine`,<br>`TextWriter.Write`, `TextWriter.WriteLine`, `Debug.WriteLine(String, Object[])`,<br>`Trace.TraceError(String, Object[])`, `Trace.TraceInformation(String, Object[])`,<br>`Trace.TraceWarning(String, Object[])` and `TraceSource.TraceInformation(String, Object[])`.

## Noncompliant Code Example


    s = string.Format("[0}", arg0);
    s = string.Format("{{0}", arg0);
    s = string.Format("{0}}", arg0);
    s = string.Format("{-1}", arg0);
    s = string.Format("{0} {1}", arg0);


## Compliant Solution


    s = string.Format("{0}", 42); // Compliant
    s = string.Format("{0,10}", 42); // Compliant
    s = string.Format("{0,-10}", 42); // Compliant
    s = string.Format("{0:0000}", 42); // Compliant
    s = string.Format("{2}-{0}-{1}", 1, 2, 3); // Compliant
    s = string.Format("no format"); // Compliant


## Exceptions

- No issue is raised if the format string is not a `const`.



    var pattern = "{0} {1} {2}";
    var res = string.Format(pattern, 1, 2); // Compliant, not const string are not recognized


- No issue is raised if the argument is not an inline creation array.



    var array = new int[] {};
    var res = string.Format("{0} {1}", array); // Compliant we don't know the size of the array


- This rule doesn't check whether the format specifier (defined after the `:`) is actually valid.


## See

- [CERT, FIO47-C.](https://www.securecoding.cert.org/confluence/x/wQA1) - Use valid format strings

