Trivial properties, which include no logic but setting and getting a backing field should be converted to auto-implemented properties, yielding
cleaner and more readable code.

## Noncompliant Code Example

<pre>
public class Car
{
  private string _make;
  public string Make // Noncompliant
  {
    get { return _make; }
    set { _make = value; }
  }
}
</pre>

## Compliant Solution

<pre>
public class Car
{
  public string Make { get; set; }
}
</pre>