The information that an enumeration type is actually an enumeration or a set of flags should not be duplicated in its name.

## Noncompliant Code Example

<pre>
enum FooFlags // Noncompliant
{
    Foo = 1
    Bar = 2
    Baz = 4
}
</pre>

## Compliant Solution

<pre>
enum Foo
{
    Foo = 1
    Bar = 2
    Baz = 4
}
</pre>