## Why is this an issue?
 
The information that an enumeration type is actually an enumeration or a set of flags should not be duplicated in its name.
 
### Noncompliant code example

    enum FooFlags // Noncompliant
    {
        Foo = 1
        Bar = 2
        Baz = 4
    }

### Compliant solution

    enum Foo
    {
        Foo = 1
        Bar = 2
        Baz = 4
    }