A method or class with too many type parameters has likely aggregated too many responsibilities and should be split.

## Noncompliant Code Example

With the default parameter value of 2:

<pre>

 <s, t, u, v>
   void foo() {} // Noncompliant; not really readable

  <string, integer, object, string>
   foo(); // especially on invocations 
  </string,>
 </s,></pre>