
A method or class with too many type parameters has likely aggregated too many responsibilities and should be split.

## Noncompliant Code Example

With the default parameter value of 2:


    &lt;S, T, U, V&gt; void foo() {} // Noncompliant; not really readable
    &lt;String, Integer, Object, String&gt;foo(); // especially on invocations

