
Correctly updating a `static` field from a non-static method is tricky to get right and could easily lead to bugs if there are multiple<br>class instances and/or multiple threads in play.

This rule raises an issue each time a `static` field is updated from a non-static method or property.

## Noncompliant Code Example


    public class MyClass
    {
      private static int count = 0;
    
      public void DoSomething()
      {
        //...
        count++;  // Noncompliant
      }
    }

