## Why is this an issue?
 
Thereâ€™s no reason to use literal boolean values in assertions. Doing so is at best confusing for maintainers, and at worst a bug.
 
### Noncompliant code example

    bool b = true;
    NUnit.Framework.Assert.AreEqual(true, b);
    Xunit.Assert.NotSame(true, b);
    Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual(true, b);
    System.Diagnostics.Debug.Assert(true);