## Why is this an issue?
 
Redundant parentheses are simply wasted keystrokes, and should be removed.
 
### Noncompliant code example

    [MyAttribute()] //Noncompliant
    class MyClass
    {
      public int MyProperty { get; set; }
      public static MyClass CreateNew(int propertyValue)
      {
        return new MyClass() //Noncompliant
        {
          MyProperty = propertyValue
        };
      }
    }

### Compliant solution

    [MyAttribute]
    class MyClass
    {
      public int MyProperty { get; set; }
      public static MyClass CreateNew(int propertyValue)
      {
        return new MyClass
        {
          MyProperty = propertyValue
        };
      }
    }