Namespaces with no lines of code clutter a project and should be removed. 

## Noncompliant Code Example

<pre>
namespace MyEmptyNamespace // Noncompliant
{

}
</pre>

## See

*   [CERT, MSC07-CPP.](https://www.securecoding.cert.org/confluence/x/OYIyAQ) - Detect and remove dead code