There's no point in checking a variable against the value you're about to assign it. Save the cycles and lines of code, and simply perform the
assignment.

## Noncompliant Code Example

<pre>
if (x != a)  // Noncompliant; why bother?
{
  x = a;
}
</pre>

## Compliant Solution

<pre>
x = a;
</pre>