
It may be a good idea to raise an exception in a constructor if you're unable to fully flesh the object in question, but not in an<br>`exception` constructor. If you do, you'll interfere with the exception that was originally being thrown. Further, it is highly unlikely<br>that an exception raised in the creation of an exception will be properly handled in the calling code, and the unexpected, unhandled exception will<br>lead to program termination.

## Noncompliant Code Example


    class MyException: Exception
    {
        public void MyException()
        {
             if (bad_thing)
             {
                 throw new Exception("A bad thing happened");  // Noncompliant
              }
        }
    }

