Creating an extension method that extends `object` is not recommended because it makes the method available on *every* type. Extensions should be applied at the most specialized level possible, and that is very unlikely to be `object`.
 
## Noncompliant Code Example

    using System;
    
    namespace MyLibrary
    {
        public static class MyExtensions
        {
            public static void Foo(this object o)  //Noncompliant
            {
                // ...
            }
        }
    }