Using upper case literal suffixes removes the potential ambiguity between "1" (digit 1) and "l" (letter el) for declaring literals.
 
## Noncompliant Code Example

    const long b = 0l;      // Noncompliant

## Compliant Solution

    const long b = 0L;